pkg/operator/controller/ingress/status.go

```diff
@@ -123,21 +127,129 @@ func computeIngressAvailableCondition(deployment *appsv1.Deployment) operatorv1.
 	}
 }
 
-func computeIngressDegradedCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
-	for _, cond := range deployment.Status.Conditions {
-		if cond.Type == appsv1.DeploymentProgressing && cond.Status == corev1.ConditionFalse && cond.Reason == "ProgressDeadlineExceeded" {
-			return operatorv1.OperatorCondition{
-				Type:    operatorv1.OperatorStatusTypeDegraded,
-				Status:  operatorv1.ConditionTrue,
-				Reason:  "DeploymentFailed",
-				Message: fmt.Sprintf("The deployment failed (reason: %s) with message: %s", cond.Reason, cond.Message),
+// computeDeploymentDegradedCondition computes the ingresscontroller's
+// "DeploymentDegraded" status condition by examining the status of
+// the deployment.
+func computeDeploymentDegradedCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
+	maxUnavailable, _ := intstr.GetValueFromIntOrPercent(deployment.Spec.Strategy.RollingUpdate.MaxUnavailable, int(*(deployment.Spec.Replicas)), false)
+	if int(deployment.Status.UnavailableReplicas) > maxUnavailable {
```

Comment:
Is this re-implementing the k8s deployment primitive's condition reporting?

I would have expected a check for this deployment condition: Available=False (MinimumReplicasUnavailable)

Reply:
Yeah, I think we can do that.  (I was inspired by your approach in cluster-dns-operator, but there, you were working with a DemonSet, and here, we have a Deployment, which has an `Available` condition.)

Reply:
Done: https://github.com/openshift/cluster-ingress-operator/compare/c6c82534a80ec899646f75c45709f8afd3567104..61f4fccd7b271c95ad91b51341b3c7c08e0a34b4

Reply:
Yeah, DameonSet doesn't give us any conditions to work with ☹️